### PR TITLE
Add support for serving the PAC file

### PR DESCRIPTION
**Pull request by luke-clifton**

### CODE REVIEW

File: internal/pac.go [hunk 1]
```diff
@@ -7,20 +7,28 @@ import (
...
+	Source  []byte
 	Parser *gopac.Parser
 	mux    sync.Mutex
 }
 
 func (pac *PAC) Init() error {
 	pac.Parser = new(gopac.Parser)
-	return pac.Parser.Parse(pac.File)
+	src, err := ioutil.ReadFile(pac.File)
+	if err != nil {
```
Comment (luke-clifton) [inline internal/pac.go:26]: I'm meant to check if `src` is `nil` actually I think?

File: internal/pac.go [hunk 2]
```diff
@@ -7,20 +7,28 @@ import (
...
+		log.Errorf("Failed to read PAC file: %s", err)
+		return nil
```
Comment (luke-clifton) [inline internal/pac.go:28]: I'm supposed to return the error here aren't I?

Comment (luke-clifton) [reply internal/pac.go:28]: Yes, I see, it's mutating the object, and the error should be returned.

File: internal/pac.go (hunk 1 above)
Comment (robbiemcmichael) [inline internal/pac.go:27]: Checking for `err != nil` is correct according to the docs: https://pkg.go.dev/io/ioutil#ReadFile

File: pkg/proxy/proxy.go [hunk 3]
```diff
@@ -42,21 +49,26 @@ func (proxy *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+		for name, values := range resp.Header {
+			w.Header()[name] = values
+		}
 
-	w.WriteHeader(resp.StatusCode)
+		w.WriteHeader(resp.StatusCode)
 
-	io.Copy(w, resp.Body)
+		io.Copy(w, resp.Body)
+	}
```
Comment (robbiemcmichael) [inline pkg/proxy/proxy.go:71]: This package meant to be used as a library, so I'd like to find a way to support serving the PAC file without adding any special cases to this.

I think it should be possible to change just the [HTTP server in `main.go`](https://github.com/robbiemcmichael/paxy/blob/b0a915bc81b15a2c04aa3c4a038a0c23f4642e91/main.go#L59) which is just used by the `paxy` executable. If we write a handler function that checks if `r.URL.Host == ""` then we should be able to send it to a special case handler that can server the PAC file.

**Pull request closed without merging**